test/plugins/unit/test_plugin_random_password_generator.py

```diff
@@ -0,0 +1,17 @@
+from tracardi.process_engine.action.v1.password_generator_action import PasswordGeneratorAction
+from tracardi.service.plugin.service.plugin_runner import run_plugin
+
+
+def test_should_return_password_length():
+    init = {"max_length": 12,
+            "min_length": 10,
+            "uppercase": 5,
+            "lowercase": 5,
+            "special_characters": 3}
+
+    payload = {}
+
+    result = run_plugin(PasswordGeneratorAction, init, payload)
+
+    assert result.output.port == "password"
+    assert (len(result.output.value["password"]) - 1) == init["max_length"]
```

Comment:
sure that line passes? It seems like the password can be anywhere between 10 and 12 characters long. I would change that to init["min_length"] <= len(result.output.value["password"]) <= init["max_length"]. Of course if these bounds are included themselves. Otherwise it's worth to mention it in the documentation.